channel/signedstate.go

```diff
@@ -1,17 +1,60 @@
 package channel
 
 import (
+	"errors"
+
 	"github.com/statechannels/go-nitro/channel/state"
 )
 
-type SignedState struct {
-	State state.VariablePart
-	Sigs  map[uint]state.Signature // keyed by participant index
+type signedState struct {
+	state state.State
+	sigs  map[uint]state.Signature // keyed by participant index
+}
+
+// newSignedState initializes a SignedState struct for the given state.
+// The signedState returned will have no signatures.
+func newSignedState(s state.State) signedState {
+	return signedState{s, make(map[uint]state.Signature, len(s.Participants))}
+}
```

Comment:
Strongly approve of use of constructor -- I will echo my question above about whether this should be exported or not (capitalised first letter).

Reply:
I tried to touch on this above but my goal was that only the `channel` package can create a new `signedState`. Any other packages can only read data from `signedState` (when it's exposed on the channel via `SignedStateForTurnNum`)